### DISCUSSION TITLE
PAGImageView使用

### DISCUSSION THREAD

**Discussion by ThinkAgains** [Tencent/libpag#1997] [Q&A] [answered]
用户可以自定义头像框动效中的文本内容，在列表中使用，PAGImageView和PagView哪一种性能会更好些？使用PAGImageView的话会不会重复缓存文件（每个不同文件都会缓存一遍？还是只缓存一次），长期使用缓存会不会增多？

**Comment by kevingpqi123** [comment, marked as answer]
具体看下这篇文章：https://pag.art/docs/use-pagimageview.html
如果会修改文本或占位图，使用 PAGImageView 会自动清除缓存数据，此时还是使用 PAGView

**Reply by kevingpqi123** [reply]
PAGImageView 缓存的 key 通过的是文件路径 + 尺寸信息 + ContentVersion，对于同一个 pag 文件，如果以上数据保持一致，只会缓存一次，否则会缓存多次，ContentVersion > 0 (修改文本或占位图), 不会持久化缓存

**Reply by ThinkAgains** [reply]
在修改文本或者占位图这种情况下（不持久缓存），列表中使用（尺寸不会太大），PagView和PagImageView哪个会更好些？

**Reply by kevingpqi123** [reply]
针对这种情况，前面已经推荐你使用 PAGView 了， PAGImageView 不持久化缓存第一次渲染的时候还是相当于要逐帧渲染，除非你这个页面会停留很久且循环滑动，缓存成功后内存会降低

**Reply by ThinkAgains** [reply]
好的谢谢
